pocs/utils/logger.py

```diff
@@ -1,45 +1,86 @@
+import os
+import sys
+import time
 import datetime
 import logging
 import logging.config
-import os
-import time
-
+from tempfile import gettempdir
 
 from .config import load_config
 
 
+class PanLogger(object):
+    """ Logger for PANOPTES with format style strings """
+
+    def __init__(self, logger):
+        super(PanLogger, self).__init__()
+        self.logger = logger
+
+    def _process_str(self, fmt, *args, **kwargs):
+        log_str = fmt
+        if len(args) > 0:
+            log_str = fmt.format(*args, **kwargs)
+
+        return log_str
+
+    def debug(self, fmt, *args, **kwargs):
+        self.logger.debug(self._process_str(fmt, *args, **kwargs))
+
+    def info(self, fmt, *args, **kwargs):
+        self.logger.info(self._process_str(fmt, *args, **kwargs))
+
+    def warning(self, fmt, *args, **kwargs):
+        self.logger.warning(self._process_str(fmt, *args, **kwargs))
+
+    def error(self, fmt, *args, **kwargs):
+        self.logger.error(self._process_str(fmt, *args, **kwargs))
+
+
 def get_root_logger(profile='panoptes', log_config=None):
     """ Creates a root logger for PANOPTES used by the PanBase object
-
     Returns:
         logger(logging.logger): A configured instance of the logger
     """
 
     # Get log info from config
     log_config = log_config if log_config else load_config('log').get('logger', {})
 
+    invoked_script = os.path.basename(sys.argv[0])
+    log_dir = '{}/logs'.format(os.getenv('PANDIR', gettempdir()))
+    log_fname = '{}-{}-{}'.format(invoked_script, os.getpid(),
+                                  datetime.datetime.utcnow().strftime('%Y%m%dT%H%M%SZ'))
+    log_symlink = '{}/{}.log'.format(log_dir, invoked_script)
+
     # Alter the log_config to use UTC times
     if log_config.get('use_utc', True):
```

Comment:
Can we apply use_utc to the file names too? I might like to run my unit with eastern time in the log files, and log file names. I certainly found during debugging the polar alignment issues that it was confusing to add utc in with the other issues.